src/test/kotlin/cacophony/controlflow/FunctionHandlerTest.kt

```diff
@@ -229,45 +229,53 @@ class FunctionHandlerTest {
             val childHandler = handlers[0]
             val parentHandler = handlers[1]
 
-            val staticLinkNode =
-                childHandler.generateCallFrom(
-                    parentHandler,
-                    emptyList(),
-                    null,
-                )[0]
+            mockkStatic(::generateCall)
+            childHandler.generateCallFrom(
+                parentHandler,
+                emptyList(),
+                null,
+                false,
+            )
 
-            // This test isn't too interesting, it's more about checking if nothing fails rather if it returns particular value.
-            val staticLinkAccess = childHandler.generateVariableAccess(childHandler.getStaticLink())
-            val expected =
-                CFGNode.Assignment(
-                    CFGNode.MemoryAccess(staticLinkAccess),
-                    CFGNode.RegisterUse(Register.FixedRegister(X64Register.RBP)),
+            verify {
+                generateCall(
+                    any(),
+                    listOf(
+                        CFGNode.RegisterUse(Register.FixedRegister(X64Register.RBP)),
+                    ),
+                    any(),
+                    false,
                 )
+            }
 
-            assertThat(staticLinkNode).isEqualTo(expected)
+            unmockkStatic(::generateCall)
         }
 
         @Test
         fun `function calling itself works`() {
             val handlers = mockFunDeclarationAndFunHandlerWithParents(0, 3)
             val childHandler = handlers[0]
 
-            val staticLinkNode =
-                childHandler.generateCallFrom(
-                    childHandler,
-                    emptyList(),
-                    null,
-                )[0]
+            mockkStatic(::generateCall)
+            childHandler.generateCallFrom(
+                childHandler,
+                emptyList(),
+                null,
+                false,
+            )
 
-            // This test isn't too interesting, it's more about checking if nothing fails rather if it returns particular value.
-            val staticLinkAccess = childHandler.generateVariableAccess(childHandler.getStaticLink())
-            val expected =
-                CFGNode.Assignment(
-                    CFGNode.MemoryAccess(staticLinkAccess),
-                    childHandler.generateVariableAccess(childHandler.getStaticLink()),
+            verify {
+                generateCall(
+                    any(),
+                    listOf(
+                        CFGNode.MemoryAccess(CFGNode.RegisterUse(Register.FixedRegister(X64Register.RBP))),
+                    ),
+                    any(),
+                    false,
                 )
+            }
 
-            assertThat(staticLinkNode).isEqualTo(expected)
+            unmockkStatic(::generateCall)
```

Comment:
Maybe extract this common test procedure to a separate function that takes two handlers and a `CFGNode`? Something like `assertThatFirstHandlerGeneratesCallFromSecondHandler`?. I think it's important to keep all calls to `mockkStatic` and `unmockkStatic` contained in one place.